src/main/java/com/fastcampus/projectboard/repository/querydsl/ArticleRepositoryCustomImpl.java

```diff
@@ -23,4 +30,41 @@ public List<String> findAllDistinctHashtags() {
                 .where(article.hashtag.isNotNull())
                 .fetch();
     }
+
+    @Override
+    public Page<Article> findBySearchKeyword(SearchType searchType, String searchKeyword, Pageable pageable) {
+        QArticle article = QArticle.article;
+
+        Predicate searchTypePredicate = getPredicateBy(searchType, searchKeyword, article);
+
+        List<Article> contents = from(article)
+                .select(article)
+                .where(searchTypePredicate)
+                .offset(pageable.getOffset())
+                .limit(pageable.getPageSize())
+                .fetch();
+
+        long count = from(article)
+                .select()
+                .where(searchTypePredicate)
+                .fetchCount();
+
+        return new PageImpl<>(contents, pageable, count);
+    }
+
+    private Predicate getPredicateBy(SearchType searchType, String searchKeyword, QArticle article) {
+        if (searchKeyword == null || searchKeyword.isBlank()) {
+            return null;
+        }
+
+        Predicate result = null;
+        switch (searchType) {
+            case TITLE -> result = article.title.contains(searchKeyword);
+            case CONTENT -> result = article.content.contains(searchKeyword);
+            case ID -> result = article.userAccount.userId.contains(searchKeyword);
+            case NICKNAME -> result = article.userAccount.nickname.contains(searchKeyword);
+            case HASHTAG -> result = article.hashtag.eq(searchKeyword);
+        }
+        return result;
+    }
```

Comment:
`null` 방어코드를 상단에 배치하면 이렇게 갈 수 있을 것 같은데요?

```suggestion
    private Predicate getPredicateBy(SearchType searchType, String searchKeyword, QArticle article) {
        if (searchType == null || searchKeyword == null || searchKeyword.isBlank()) {
            return null;
        }

        return switch (searchType) {
            case TITLE		-> article.title.contains(searchKeyword);
            case CONTENT	-> article.content.contains(searchKeyword);
            case ID		-> article.userAccount.userId.contains(searchKeyword);
            case NICKNAME	-> article.userAccount.nickname.contains(searchKeyword);
            case HASHTAG	-> article.hashtag.eq(searchKeyword);
        }
    }
```

Reply:
아! 자바 12부터 switch문에서 반환이 가능했죠 해당 내용이 더 좋아보이네요 감사합니다.

Reply:
Commit suggestion을 사용하면 conversation이 닫히는군요